Source/Core/Core/IOS/ES/Formats.cpp

```diff
@@ -303,32 +277,47 @@ std::vector<u8> TicketReader::GetRawTicketView(u32 ticket_num) const
 
 u32 TicketReader::GetDeviceId() const
 {
-  return Common::swap32(m_bytes.data() + GetOffset() + offsetof(Ticket, device_id));
+  return Common::swap32(m_bytes.data() + offsetof(Ticket, device_id));
 }
 
 u64 TicketReader::GetTitleId() const
 {
-  return Common::swap64(m_bytes.data() + GetOffset() + offsetof(Ticket, title_id));
+  return Common::swap64(m_bytes.data() + offsetof(Ticket, title_id));
 }
 
 std::vector<u8> TicketReader::GetTitleKey() const
 {
   u8 iv[16] = {};
-  std::copy_n(&m_bytes[GetOffset() + offsetof(Ticket, title_id)], sizeof(Ticket::title_id), iv);
-  auto common_key_handle = m_bytes.at(GetOffset() + offsetof(Ticket, common_key_index)) == 0 ?
+  std::copy_n(&m_bytes[offsetof(Ticket, title_id)], sizeof(Ticket::title_id), iv);
+  auto common_key_handle = m_bytes.at(offsetof(Ticket, common_key_index)) == 0 ?
                                HLE::IOSC::HANDLE_COMMON_KEY :
                                HLE::IOSC::HANDLE_NEW_COMMON_KEY;
 
   std::vector<u8> key(16);
   HLE::IOSC iosc;
-  iosc.Decrypt(common_key_handle, iv, &m_bytes[GetOffset() + offsetof(Ticket, title_key)], 16,
-               key.data(), HLE::PID_ES);
+  iosc.Decrypt(common_key_handle, iv, &m_bytes[offsetof(Ticket, title_key)], 16, key.data(),
+               HLE::PID_ES);
   return key;
 }
 
+void TicketReader::DeleteTicket(u64 ticket_id_to_delete)
+{
+  std::vector<u8> new_ticket;
+  const size_t num_tickets = GetNumberOfTickets();
+  for (size_t i = 0; i < num_tickets; ++i)
+  {
+    const auto ticket_start = m_bytes.cbegin() + sizeof(Ticket) * i;
+    const u64 ticket_id = Common::swap64(&*ticket_start + offsetof(Ticket, ticket_id));
```

Comment:
Are you sure this does the right thing? From staring at it on the GitHub diff, this swap64's the address of the ticket id, not the value of it?

This works in other places because they expect an address, but swap64 wants the value...